service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceMethod.java

```diff
@@ -348,7 +348,19 @@ public Job scheduleDeviceMethod(
             throw new IllegalArgumentException("negative maxExecutionTimeInSeconds");
         }
 
-        Job job = new Job(this.getAuthenticationToken());
+        Job job;
+        if (this.credential != null)
+        {
+            job = new Job(this.hostName, this.credential);
```

Comment:
is this the TokenCredential type? can we name it as such?

Reply:
From previous discussions with Sindhu and David, we agreed to call the TokenCredential argument "credential" to match the naming within the azure identity library itself